### ISSUE TITLE
Clean old dependencies from system

### ISSUE THREAD

**Issue by krisgesling** [MycroftAI/mycroft-skills-manager#86]
When a Skill is installed it can define system packages, Python packages and other Mycroft Skills as dependencies. These are installed automatically but not removed if the Skill is uninstalled at a later date.

We cannot just remove any dependencies on uninstall as these may be required by other Skills or services. 

One suggestion from https://github.com/MycroftAI/skill-date-time/issues/90 is to perform some type of system cleaning at regular intervals such as major releases. This may be possible on a Mycroft only system, but may be more challenging on a shared system like a desktop install unless its a Snap container or similar. Even if we only cleaned Python packages from the Mycroft venv it would be an improvement on the current system.

**Comment by forslund** [comment]
For a desktop install there are two ways to go. virtual-environment like in [mycroft-desktop-deb](https://github.com/forslund/mycroft-desktop-deb) or the skills are installed using the system package manager like the implementation @puretryout has made.

In the first case the major version cleanup will work. In the second the system package manger will manage the dependencies so it's a non-issue.

**Comment by PureTryOut** [comment]
Indeed, when managed with a system package manager this is a non-issue. I hope that use-case will remain supported and some automatic cleanup by msm can be disabled in some way.

**Comment by forslund** [comment]
Thinking around it. msm would only do cleanup when doing an upgrade or a remove and not automatically. The mycroft package for the Mark-X's or the desktop-deb would handle any scheduled cleanup so it shouldn't affect the installs handled by distro package managers. (you should be safe)

A possibility is to have a `msm clean` command. Parse the skills requirements and add the ones by core then remove any package that isn't on that list or required by a package on that list. A bit of tree-building but it should be doable.
